pkg/imgutil/dockerconfigresolver/credentialsstore_test.go

```diff
@@ -0,0 +1,384 @@
+					"http://someregistry": {
+						"username": "username"
+					}
+				}
+			}`
+				return writeContent(t, content)
+			},
+		},
+		{
+			description: "Reading from legacy / alternative identifiers: https://someregistry",
+			username:    "username",
+			setup: func() string {
+				content := `{
+				"auths": {
+					"https://someregistry": {
+						"username": "username"
+					}
+				}
+			}`
+				return writeContent(t, content)
+			},
+		},
+		{
+			description: "Reading from legacy / alternative identifiers: http://someregistry:443",
+			username:    "username",
+			setup: func() string {
+				content := `{
+				"auths": {
+					"http://someregistry:443": {
+						"username": "username"
+					}
+				}
+			}`
+				return writeContent(t, content)
+			},
+		},
+		{
+			description: "Reading from legacy / alternative identifiers: https://someregistry:443",
+			username:    "username",
+			setup: func() string {
+				content := `{
+				"auths": {
+					"https://someregistry:443": {
+						"username": "username"
+					}
+				}
+			}`
+				return writeContent(t, content)
+			},
+		},
+		{
+			description: "Canonical form is preferred over legacy forms",
+			username:    "pick",
+			setup: func() string {
+				content := `{
+	"auths": {
+		"http://someregistry:443": {
+			"username": "ignore"
+		},
+		"https://someregistry:443": {
+			"username": "ignore"
+		},
+		"someregistry": {
+			"username": "ignore"
+		},
+		"someregistry:443": {
+			"serveraddress": "bla",
+			"username": "pick"
+		},
+		"http://someregistry": {
+			"username": "ignore"
+		},
+		"https://someregistry": {
+			"username": "ignore"
+		}
+	}
+}`
+				return writeContent(t, content)
+			},
+		},
+	}
+
+	t.Run("Working credentials store", func(t *testing.T) {
+
+		for _, tc := range testCases {
+			t.Run(tc.description, func(t *testing.T) {
+				registryURL, err := Parse("someregistry")
+				if err != nil {
+					t.Fatal(err)
+				}
+				cs, err := NewCredentialsStore(tc.setup())
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				var af *Credentials
+				af, err = cs.Retrieve(registryURL, true)
+				assert.ErrorIs(t, err, nil)
+				assert.Equal(t, af.Username, tc.username)
+				assert.Equal(t, af.ServerAddress, "someregistry:443")
+				assert.Equal(t, af.Password, tc.password)
+			})
+		}
+	})
+
+	t.Run("Namespaced host", func(t *testing.T) {
+		server := "somehost.com/path?ns=someregistry.com"
+		registryURL, err := Parse(server)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		content := `{
+				"auths": {
+					"nerdctl-experimental://someregistry.com:443/host/somehost.com:443/path": {
+						"username": "username"
+					}
+				}
+			}`
+		dir := writeContent(t, content)
+		cs, err := NewCredentialsStore(dir)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var af *Credentials
+		af, err = cs.Retrieve(registryURL, true)
+		assert.ErrorIs(t, err, nil)
+		assert.Equal(t, af.Username, "username")
+		assert.Equal(t, af.ServerAddress, "somehost.com:443/path?ns=someregistry.com")
```

Comment:
`example.com` is recommended
https://www.iana.org/help/example-domains

Reply:
Will do.